client-http/src/main/java/com/cloudera/livy/client/http/LivyConnection.java

```diff
@@ -90,24 +91,40 @@ public boolean isContentCompressionEnabled() {
       }
     };
 
-    Credentials dummyCredentials = new Credentials() {
-      @Override
-      public String getPassword() {
-        return null;
+    Credentials credentials;
+    // If user info is specified in the url, pass them to the CredentialsProvider.
+    if (uri.getUserInfo() != null) {
+      String[] userInfo = uri.getUserInfo().split(":");
```

Comment:
Do you enforce user to have both username and password in URL. As I know only username is acceptable in URL format. Should we also consider it?